pandas/tests/indexes/numeric/test_setops.py

```diff
@@ -74,6 +74,17 @@ def test_range_float_union_dtype(self):
         result = other.union(index)
         tm.assert_index_equal(result, expected)
 
+    def test_range_uint64_union_dtype(self):
```

Comment:
this behavior looks the same on main.  how does the test relate to the PR?

Reply:
(im ok adding the test, just want to understand the connection)

Reply:
Sure: when working on this PR I tried some changes in `_find_common_type_compat` and `find_result_type` that were causing the union dtype to be different from object, but the error wasn't detected by any test, so I added this one.
The method `_find_common_type_compat` is called in `Index.union` here: https://github.com/pandas-dev/pandas/blob/c92b3701efff16293bd110a3f06f458040b43d97/pandas/core/indexes/base.py#L3316